common/src/main/java/com/redhat/lightblue/rest/audit/LightblueAuditServletFilter.java

```diff
@@ -0,0 +1,527 @@
+
+
+            HttpServletResponse httpServletResponse = (HttpServletResponse) res;
+            HttpServletResponse httpServletResponseWrapperBuffered = new HttpServletResponseWrapperBuffered(httpServletResponse,new ByteArrayPrintWriter());
+            res = httpServletResponseWrapperBuffered;
+        }
+
+        if(auditReqFlag) {
+            stopwatch = Stopwatch.createStarted();
+        }
+
+        fChain.doFilter(hReq, res);
+
+        if(auditReqFlag) {
+            stopwatch.stop();
+            long elapsedTime = stopwatch.elapsed(TimeUnit.NANOSECONDS); // elapsedTime in ns
+            logEntryBuilder.setTimeElapsedInNs(elapsedTime);
+            HttpServletResponseWrapperBuffered httpServletResponseWrapperBuffered = (HttpServletResponseWrapperBuffered) res;
+            httpServletResponseWrapperBuffered.byteArrayPrintWriter.printWriter.flush();
+            byte[] bytes = httpServletResponseWrapperBuffered.byteArrayPrintWriter.toByteArray();
+            res.getOutputStream().write(bytes);
+            logEntryBuilder.setResponseSize(bytes.length);
+            final LogEntry logEntry = logEntryBuilder.createLogEntry();
+            String logEntryString = String.format(
+                    "Audited lightblue rest request => " +
+                            "{ " +
+                                "\"initialTimestamp\":\"%s\", " +
+                                "\"currentTimestamp\":\"%s\" , " +
+                                "\"principal\":\"%s\" , " +
+                                "\"resource\":\"%s\" , " +
+                                "\"operation\":\"%s\" , " +
+                                "\"entityName\":\"%s\" , " +
+                                "\"entityVersion\":\"%s\" , " +
+                                "\"entityStatus\":\"%s\" , " +
+                                "\"requestSize\":\"%d\" , " +
+                                "\"responseSize\":\"%d\" , " +
+                                "\"timeElapsedInNs\":\"%d\"  " +
+                            " }",
+                    logEntry.getTimestampText(),
+                    threadDateFormat.get().format(new Date()),
+                    logEntry.getPrincipal().getName(),
+                    logEntry.getResource(),
+                    logEntry.getOperation(),
+                    logEntry.getEntityName(),
+                    logEntry.getEntityVersion(),
+                    logEntry.getEntityStatus(),
+                    logEntry.getRequestSize(),
+                    logEntry.getResponseSize(),
+                    logEntry.getTimeElapsedInNs()
+            );
+            LOGGER.info(logEntryString);
+        }
+
+        if(LOGGER.isDebugEnabled()) {
+            LOGGER.debug("LightblueAuditServletFilter.doFilter invoked - end");
+        }
+    }
+
+    protected void setOperationEnittyVersionStatus(HttpServletRequest hReq, boolean isMetadata, LogEntryBuilder logEntryBuilder) {
```

Comment:
Not sure if I got the idea you mentioned about the single instance of Matcher. 

Do you mean create an variable for each case up front (like, before the switch statement) so then I can use 'else if' statement?  In this case, I`m a bit concerned about it could create unnecessary Matcher objects as many cases would not be evaluated during a single request processing (which would give more work to the GC after some requests) , specially as this filter will handle both the data and metadata requests. If GC pause/impact is not a problem, I don`t mind we change to this way.